### PR TITLE
rke2-monitoring

### PR DESCRIPTION
**Pull request by MonzElmasry**
Enable the rancher dashboard to monitor rke2 clusters by adding support for `rke2ControllerManager`, `rke2Scheduler`, `rke2Proxy`, and `rke2Etcd`
issue :https://github.com/rancher/rke2/issues/33
canal and nginx will be supported in a different chart.

### CODE REVIEW

**Comment by cjellick** [comment]
Why is there so much stuff changing in this PR that is unrelated to your exact change? We should strive to make these patch PRs as small as possible

**Comment by MonzElmasry** [comment]
@cjellick looks like blocks of code are moving around in the patch file when i make a patch, these changes are generated

**Comment by aiyengar2** [comment]
> @cjellick looks like blocks of code are moving around in the patch file when i make a patch, these changes are generated

Hmm this is weird, I haven't experienced code blocks moving around when trying to generate changes on the patch before. Let's take a look at this when we meet later today @MonzElmasry

**Comment by StrongMonkey** [comment]
I don't have to a good way to fix this right now. It appears that changes are introduced because of the diff on GNU and BSD produces different output. By taking close look at changes, it is swapping the positions so nothing critically gets changed.

**Comment by MonzElmasry** [comment]
https://github.com/rancher/charts/compare/dev-v2.5...MonzElmasry:new-something
generated changes on make patch

File: packages/rancher-monitoring/rancher-monitoring.patch [hunk 1]
```diff
@@ -1657,6 +1668,60 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/val
...
++      - effect: "NoSchedule"
++        operator: "Exists"
++
++rke2Proxy:
++  enabled: false
++  metricsPort: 10249
++  component: kube-proxy
++  clients:
++    port: 10013
++    useLocalhost: true
```
Comment (cbron) [inline packages/rancher-monitoring/rancher-monitoring.patch:1710]: Do we not need the tolerations settings here ?

Comment (MonzElmasry) [reply packages/rancher-monitoring/rancher-monitoring.patch:1709]: In my testing it didnt need any, but added anyway just in case

Comment (cbron) [reply packages/rancher-monitoring/rancher-monitoring.patch:1709]: @brandond or @cjellick do we know which parts we do or don't want tolerations on ?

Comment (brandond) [reply packages/rancher-monitoring/rancher-monitoring.patch:1709]: Are we talking about the bootstrap flag on the HelmChart that adds tolerations? If so, I think we should only put those on things that need to run as part of cluster bootstrap. Other stuff like monitoring can wait until after the core functionality is up.

Comment (brandond) [reply packages/rancher-monitoring/rancher-monitoring.patch:1709]: Oh, I see. The these are just to let monitoring run on masters or nodes that have been tainted NoExecute/NoSchedule? Seems fine to me.

Comment (cbron) [reply packages/rancher-monitoring/rancher-monitoring.patch:1709]: Ok, if we are good with that. I was more concerned about users being able to edit this setting via values.

**Comment by cjellick** [comment]
🚢 🇮🇹

**Pull request merged**